File: cache/play-jcache/src/main/java/play/libs/jcache/JCacheComponents.java
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package play.libs.jcache;

import javax.cache.CacheManager;
import javax.cache.Caching;
import play.Environment;

/** JCache components */
public interface JCacheComponents {

  Environment environment();

  default CacheManager cacheManager() {
    return Caching.getCachingProvider(environment().classLoader()).getCacheManager();
  }
}

File: cache/play-jcache/src/main/scala/play/api/libs/jcache/JCacheComponents.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.libs.jcache

import javax.cache.CacheManager
import javax.cache.Caching

import play.api.Environment

/**
 * Components for JCache CacheManager
 */
trait JCacheComponents {
  def environment: Environment

  lazy val cacheManager: CacheManager = Caching.getCachingProvider(environment.classLoader).getCacheManager
}
